pkg/prom/ha/api.go

```diff
@@ -0,0 +1,187 @@
+package ha
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/pkg/agentproto"
+	"github.com/grafana/agent/pkg/prom/ha/configapi"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// APIHandler is a function that returns a configapi Response type
+// and optionally an error.
+type APIHandler func(r *http.Request) (interface{}, error)
+
+// API can wire HTTP and gRPC routes.
+type API struct {
+	mut         sync.Mutex
+	server      *Server
+	logger      log.Logger
+	scrapingApi *deferredScrapingServiceServer
+}
```

Comment:
I'm not really happy with the need to do this. I need to think about updating config within a Server instead.